core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -219,6 +220,9 @@ class HadoopRDD[K, V](
       val bytesReadCallback = if (split.inputSplit.value.isInstanceOf[FileSplit]) {
         SparkHadoopUtil.get.getFSBytesReadOnThreadCallback(
           split.inputSplit.value.asInstanceOf[FileSplit].getPath, jobConf)
+      } else if (split.inputSplit.value.isInstanceOf[CombineFileSplit]) {
+        SparkHadoopUtil.get.getFSBytesReadOnThreadCallback(
+          split.inputSplit.value.asInstanceOf[CombineFileSplit].getPath(0), jobConf)
```

Comment:
The issue is that those are actually two different classes.  There's a CombineFileSplit for the old MR API (used by HadoopRDD) and a CombineFileSplit for the new one (used by NewHadoopRDD).